src/main/java/com/puppycrawl/tools/checkstyle/XpathFileGeneratorAuditListener.java

```diff
@@ -108,6 +108,12 @@ public void addError(AuditEvent event) {
             writer.print(checkName);
             writer.println(QUOTE_CHAR);
 
+            if (event.getModuleId() != null) {
+                writer.print("       id=\"");
+                writer.print(event.getModuleId());
+                writer.println(QUOTE_CHAR);
+            }
```

Comment:
@romani Should we still print check name if module id is populated?
We don't even display check name in violation if id is given.

Reply:
Yes, it will be good to print either ID or Check name.
The same as our cli output does.

Reply:
done